modules/accounts.js

```diff
@@ -276,83 +241,61 @@ Accounts.prototype.shared = {
 		});
 	},
 
-	getDelegatesFee: function (req, cb) {
-		return setImmediate(cb, null, {fee: constants.fees.delegate});
-	},
-
-	getAccount: function (req, cb) {
-		library.schema.validate(req.body, schema.getAccount, function (err) {
+	getAccounts: function (req, cb) {
+		library.schema.validate(req.body, schema.getAccounts, function (err) {
 			if (err) {
-				return setImmediate(cb, err[0].message);
-			}
-
-			if (!req.body.address && !req.body.publicKey) {
-				return setImmediate(cb, 'Missing required property: address or publicKey');
-			}
-
-			// self.getAccount can accept publicKey as argument, but we also compare here
-			// if account publicKey match address (when both are supplied)
-			var address = req.body.publicKey ? self.generateAddressByPublicKey(req.body.publicKey) : req.body.address;
-			if (req.body.address && req.body.publicKey && address !== req.body.address) {
-				return setImmediate(cb, 'Account publicKey does not match address');
+				return setImmediate(cb, new ApiError(err[0].message, apiCodes.BAD_REQUEST));
 			}
 
-			self.getAccount({ address: address }, function (err, account) {
+			library.logic.account.getAll(req.body, function (err, accounts) {
 				if (err) {
-					return setImmediate(cb, err);
+					return setImmediate(cb, new ApiError(err, apiCodes.BAD_REQUEST));
 				}
 
-				if (!account) {
-					return setImmediate(cb, 'Account not found');
-				}
+				var lastBlock = modules.blocks.lastBlock.get();
+				var totalSupply = __private.blockReward.calcSupply(lastBlock.height);
+
+				accounts = _.map(accounts, function (account) {
+					var delegate = null;
+
+					// Only create delegate properties if account has a username
+					if (account.username) {
+						var approval = (account.vote / totalSupply) * 100;
```

Comment:
Didn't we discuss that those fields would be gathered from `mem_accounts` table?

Reply:
If not then please move the functionality to `logic.account` because the logic would have to be shared with `delegates` module as well.

Reply:
Okay, makes sense.